src/umpire/ResourceManager.cpp

```diff
@@ -291,14 +291,14 @@ ResourceManager::hasAllocator(void* ptr)
   return m_allocations.contains(ptr);
 }
 
-void ResourceManager::registerAllocation(void* ptr, util::AllocationRecord* record)
+void ResourceManager::registerAllocation(void* ptr, util::AllocationRecord record)
```

Comment:
@jdahm, @davidbeckingsale, - just checking for babies in bathwater here...  What was the original intention behind previously passing around a pointer to an AllocationRecord instead of by value?

Reply:
The need for speed....

Reply:
Thanks @davidbeckingsale, but I don't understand.  Are you saying that we needed to pass the pointers around for speed which would imply that copying values around might be slower?

Reply:
I think the intent was just that. Copying a pointer is 8 bytes, but AllocationRecord has 3 8 byte members. @davidbeckingsale and I were deciding between speed and complexity of the memory management. Storing and passing around pointers means that the method creating the allocation records outside the map "owned" the allocation/deallocation of the structs. With increased optimizations elsewhere, I figured we could back off and switch to pass by value, simplifying the ownership model. Also, there may be certain optimizations performed for passing by value.